File: atCoder/abc/187/b/main.py
n = int(input())
pts=[]
for i in range(n): 
  x,y=[int(v) for v in input().split()]
  pts.append((x,y))

ans = 0
for i in  range(n-1):
  (xi, yi) = pts[i]
  for j in range(i+1,n):
    (xj, yj) = pts[j]
    if abs(yi - yj)>abs(xi - xj): continue
    ans += 1

print(ans)

File: atCoder/abc/187/b/main.jl
let
  v = parse.(Int, read(stdin, String)|>split)
  n = v[1]
  pts = reshape( v[2:end], (2,n))
  ans = 0
  for i in 1:n-1
    xi, yi = pts[:,i]
    for j in i+1:n
      (abs(yi - pts[2,j])>abs(xi - pts[1,j])) && continue
      ans += 1
    end
  end
  println(ans)
end
